### PR TITLE
[plugin.video.embycon] 1.5.30

### PR DESCRIPTION
**Pull request by faush01**
### Description

July release:
- Fix a few bugs
- Remove some redundant code
- Refactor some code for performance

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.embycon/resources/lib/context_monitor.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    stop_thread = False
+
+    def run(self):
+
+        item_id = None
+        log.debug("ContextMonitor Thread Started")
+        context_up = False
+        is_embycon_item = False
+
+        while not xbmc.abortRequested and not self.stop_thread:
```
Comment (enen92) [inline plugin.video.embycon/resources/lib/context_monitor.py]: `xbmc.Monitor().abortRequested()` please

File: plugin.video.embycon/resources/lib/functions.py [hunk 2]
```diff
@@ -936,27 +988,61 @@ def searchResults(params):
...
+            # set content type
+            xbmcplugin.setContent(handle, content_type)
+
+            dir_items, detected_type = processDirectory(details_result, progress, params)
+            if dir_items is not None:
+                xbmcplugin.addDirectoryItems(handle, dir_items)
+                xbmcplugin.endOfDirectory(handle, cacheToDisc=False)
+
+        elif not query_string:
+            xbmcgui.Dialog().ok("No Matches", "No items match your search.")
```
Comment (enen92) [inline plugin.video.embycon/resources/lib/functions.py]: minor: can you move this to localized strings?

File: plugin.video.embycon/resources/lib/menu_functions.py [hunk 3]
```diff
@@ -853,13 +853,13 @@ def showWidgets():
...
 
-def showSearch():
+def show_search():
     addMenuDirectoryItem(i18n('movies'), 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=Movie')
     addMenuDirectoryItem(i18n('tvshows'), 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=Series')
     addMenuDirectoryItem(i18n('episodes'), 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=Episode')
     addMenuDirectoryItem("Song", 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=Audio')
     addMenuDirectoryItem("Album", 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=MusicAlbum')
-    #addMenuDirectoryItem("Artist", 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=MusicArtists')
+    addMenuDirectoryItem("Person", 'plugin://plugin.video.embycon/?mode=NEW_SEARCH&item_type=Person')
```
Comment (enen92) [inline plugin.video.embycon/resources/lib/menu_functions.py]: Same with these items if possible

File: plugin.video.embycon/resources/lib/context_monitor.py (hunk 1 above)
Comment (faush01) [reply plugin.video.embycon/resources/lib/context_monitor.py]: This was picked up in the initial submission as well, in the other service loop I have, I did try to switch over at that time but found a bug in Kodi
Notes from my last attempt to use the monitor.

# monitor.abortRequested() is causes issues, it currently triggers for all addon cancelations which causes
# the service to exit when a user cancels an addon load action. This is a bug in Kodi.
# I am switching back to xbmc.abortRequested approach until kodi is fixed or I find a work around

Here is the forum thread on it
https://forum.kodi.tv/showthread.php?tid=319161&highlight=abortRequested

I am not sure this was pack ported to Kodi 17 yet however to maintain backward compatibility with all version of 17.x of Kodi I think I should stick with my current implementation for now and switch to the new one in Kodi 18.

Is that ok?

File: plugin.video.embycon/resources/lib/functions.py (hunk 2 above)
Comment (faush01) [reply plugin.video.embycon/resources/lib/functions.py]: will do

File: plugin.video.embycon/resources/lib/menu_functions.py (hunk 3 above)
Comment (faush01) [reply plugin.video.embycon/resources/lib/menu_functions.py]: will do

File: plugin.video.embycon/resources/lib/context_monitor.py (hunk 1 above)
Comment (faush01) [reply plugin.video.embycon/resources/lib/context_monitor.py]: @enen92 
Just wondering if this was ok.
Having to support multiple version of Kodi 17 with the same code base is my priority now.
I am not even sure how I would support addons for different versions of Kodi 17.x

Comment (enen92) [reply plugin.video.embycon/resources/lib/context_monitor.py]: Oh ok I am using v18 for a while and I was not aware of this issue in Krypton. Please make sure you adapt if you add this addon to the Leia repo at some point. Thanks

**Pull request merged**